### DISCUSSION TITLE
[v4]  Content Discovery

### DISCUSSION THREAD

**Discussion by xcrap** [tailwindlabs/tailwindcss#13824] [General]
I wanted to try the v4 alpha, but I'm very suspicious about the content discovery. I'm a web developer, and I only create custom websites where I write my own PHP/JS/TAILWIND, both frontend and backend.

For example, for Project X, I have the content set to: ["./.php", "./sources/.js", "./inc/.php", "./lang/.php"]. This is project-based, but in this project, I have an /admin/ folder where I have different content arguments because it's a back office that's independent of the main website (also running Tailwind there). It's like a subproject within the website, but it is independent. I am wondering how the automatic content discovery would work for these kinds of projects where I need to discover files in very specific locations and not anywhere else.

I was thinking a lot about this and I'm sure automatic content discovery would work for a lot of projects but without a config file with the content discovery it would be a mess for many others, all of mine for example. Can someone enlighten me on this?

**Comment by phenomen** [comment]
I also have an issue with automatic content detection. I use Blade (`.blade`) and Handlebars (`.hbs`) templates in various projects, which are not in the `src` directory but Vite's `public` directory (since those templates are not transformed or pre-processed by Vite in any way). In Tw3 it was trivial to specify my templates in the config but Tw4 completely ignores Vite `public` directory because those files are not in the Vite's module graph.
